addon/-private/system/internal-model-map.ts

```diff
@@ -24,19 +37,19 @@ export default class InternalModelMap {
    * @param id {String}
    * @return {InternalModel}
    */
-  get(id) {
+  get(id: string): InternalModel|undefined {
     return this._idToModel[id];
   }
 
-  has(id) {
+  has(id: string): boolean {
     return !!this._idToModel[id];
   }
 
-  get length() {
+  get length(): number {
```

Comment:
If you want to use tslint to enforce explicit return types, the rule looks like

```json
"rules": {
  "typedef": [true, "call-signature"]
}
```

Reply:
anything in the "rules" object applies only to TS, and there's a "jsRules" for JS

Reply:
ooooooh :D then yes, I'll turn that on :D